src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java

```diff
@@ -241,11 +254,13 @@ void handleFrame(WebSocketFrameInternal frame) {
   }
 
   private class FrameAggregator implements Handler<WebSocketFrameInternal> {
-
     private Handler<String> textMessageHandler;
     private Handler<Buffer> binaryMessageHandler;
+    private Handler<Buffer> pongHandler;
+
     private Buffer textMessageBuffer;
     private Buffer binaryMessageBuffer;
+    private Buffer pongBuffer;
```

Comment:
can pong frames be fragmented ?

Reply:
I don't think they actually can.  I'll be honest, I copy pasta'd this code to create the pongBuffer. I'll look and see if I can test anyway, or make changes if requested.

Per RFC 6455 5.5:

```
   Control frames are used to communicate state about the WebSocket.
   Control frames can be interjected in the middle of a fragmented
   message.

   All control frames MUST have a payload length of 125 bytes or less
   and MUST NOT be fragmented.
```

Reply:
I can remove the code, but I think I should throw an exception if the ping/pong frame being sent exceeds `WebSocketImplBase.maxWebSocketFrameSize`

Reply:
maybe it would be appropriate to also assert that the ping/pong frame size is less than 126 bytes.

Reply:
I don't know what's best for now, but either way should have test :-)

Reply:
I've been testing this, It seems like Netty won't allow me to send a ping frame with more than 125 bytes of data.

Example test:
```  @Test
  public void testServerWebsocketReceivedPingExceeds125Bytes() {
    String pingBody = randomAlphaString(125);//fails if this is larger 
    Buffer pongMessage = Buffer.buffer(pingBody.length());
    Integer maxFrameSize = 256;
    server = vertx.createHttpServer(new HttpServerOptions().setIdleTimeout(1).setPort(HttpTestBase.DEFAULT_HTTP_PORT).setHost(HttpTestBase.DEFAULT_HTTP_HOST).setMaxWebsocketFrameSize(maxFrameSize));
    server.websocketHandler(ws -> {
      ws.pongHandler(pong -> {
        pongMessage.setBuffer(0, pong);
        ws.close();
      });
      ws.writeFrame(WebSocketFrame.pingFrame(Buffer.buffer(pingBody)));
    }).listen(ar -> {
      client.websocket(HttpTestBase.DEFAULT_HTTP_PORT, HttpTestBase.DEFAULT_HTTP_HOST, "/", ws -> {});
    });

    Long start = System.currentTimeMillis();
    while(System.currentTimeMillis() - start < 2000 && !pingBody.equals(pongMessage.toString())) { }
    assertEquals(pingBody, pongMessage.toString());
  }```

The "writeFrame" method appears to complete successfully, but the vertxHandler method on the client side is to  be invoked.  I haven't looked into the netty source yet, but my hunch is that the behavior is from netty and not from vertx

Reply:
so we should not use a pongBuffer aggregation I think

Reply:
I'll have a slew of unit tests and error handling this evening.